app/share.tsx

```diff
@@ -89,115 +70,68 @@ export const App = ({ root }: any) => (
 	</Stack.Navigator>
 );
 
-class Root extends React.Component<{}, IState> {
-	private mounted = false;
-
-	constructor(props: any) {
-		super(props);
-		const { width, height, scale, fontScale } = Dimensions.get('screen');
-		const theme = initialTheme();
-		this.state = {
-			theme: getTheme(theme),
-			themePreferences: theme,
-			root: '',
-			width,
-			height,
-			scale,
-			fontScale
-		};
-		this.init();
-	}
+const { width, height, scale, fontScale } = Dimensions.get('screen');
+const theme = getTheme(initialTheme());
```

Comment:
Why are these properties that are only used inside Root outside of it?
As it is here, they are allocated in memory while the app is running, and not just at screen execution time.

Reply:
@GleidsonDaniel this was discussed with @diegolmello 

https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/4417#discussion_r942735088